modules/gdscript/language_server/gdscript_language_protocol.cpp

```diff
@@ -153,61 +188,59 @@ void GDScriptLanguageProtocol::initialized(const Variant &p_params) {
 }
 
 void GDScriptLanguageProtocol::poll() {
-	server->poll();
+	if (server->is_connection_available()) {
+		on_client_connected();
+	}
+	int *queue_for_discon = (int *)memalloc(sizeof(int) * clients.size());
+	int discons = 0;
+	for (int c_ind = 0; c_ind < clients.size(); c_ind++) {
+		Ref<StreamPeerTCP> peer = clients.get(c_ind);
+		StreamPeerTCP::Status status = peer->get_status();
+		if (status == StreamPeerTCP::STATUS_NONE || status == StreamPeerTCP::STATUS_ERROR) {
+			queue_for_discon[discons] = c_ind;
+			discons++;
+		} else {
+			int bytes = peer->get_available_bytes();
+			if (bytes > 0) {
+				on_data_received(c_ind, bytes);
+			}
+		}
+	}
+	for (int i = 0; i < discons; i++) {
+		on_client_disconnected(queue_for_discon[i]);
+	}
+	memfree(queue_for_discon);
 }
 
 Error GDScriptLanguageProtocol::start(int p_port, const IP_Address &p_bind_ip) {
 	if (server == NULL) {
-		server = dynamic_cast<WebSocketServer *>(ClassDB::instance("WebSocketServer"));
+		server = dynamic_cast<TCP_Server *>(ClassDB::instance("TCP_Server"));
 		ERR_FAIL_COND_V(!server, FAILED);
-		server->set_buffers(8192, 1024, 8192, 1024); // 8mb should be way more than enough
-		server->connect("data_received", this, "on_data_received");
-		server->connect("client_connected", this, "on_client_connected");
-		server->connect("client_disconnected", this, "on_client_disconnected");
 	}
-	server->set_bind_ip(p_bind_ip);
-	return server->listen(p_port);
+	return server->listen(p_port, p_bind_ip);
 }
 
 void GDScriptLanguageProtocol::stop() {
-	const int *ptr = clients.next(NULL);
-	while (ptr) {
-		clients.get(*ptr)->close();
-		ptr = clients.next(ptr);
+	for (int c_ind = 0; c_ind < clients.size(); c_ind++) {
+		Ref<StreamPeerTCP> peer = clients.get(c_ind);
+		peer->disconnect_from_host();
 	}
-	server->stop();
```

Comment:
Stop the tcp server here.

Reply:
The diff is a bit weird but I stop it on 299